popgen/mcmcSelection.go

```diff
@@ -122,13 +123,28 @@ func MetropolisHastings(data AFS, muZero float64, sigmaZero float64, iterations
 	}
 
 	maxN := findMaxN(data)
-	binomMap := make([][]float64, maxN+1)
+	allN := findAllN(data)
+	alleleFrequencyCache = cacheIndexToAlleleFrequencyCache(bins)
+	nkpCache := make([][][]float64, maxN+1)
+	var coefficient float64
+
+	for n := 0; n < len(allN); n++ {
+		nkpCache[allN[n]] = make([][]float64, allN[n])
+		for k := 0; k < len(allN[n]); k++ {
+			coefficient = numbers.BinomCoefficient(n, k)
+			nkpCache[allN[n]][k] = make([]float64, bins+1)
+			for p := 0; p < len(bins); p++ {
```

Comment:
Should it be len(bins) + 1?

Reply:
bins + 1 in fact, bins is already an int, so the len() was erroneous.